README.md

```diff
@@ -1,2 +1,109 @@
 # didcomm-python
-basic DIDComm v2 support in python
+
+Basic [DIDComm v2](https://identity.foundation/didcomm-messaging/spec) support in python.
+
+Requires Python >= 3.7.
+
+## Examples
+
+See [demo scripts](tests/demo) for details.
+
+A general usage of the API is the following:
+- Sender Side:
+  - Build a `Message` (plaintext, payload).
+  - Convert a message to a DIDComm Message for further transporting by calling one of the following:
+     - `pack_encrypted` to build an Encrypted DIDComm message
+     - `pack_signed` to build a Signed DIDComm message
+     - `pack_plaintext` to build a Plaintext DIDComm message
+- Receiver side:
+  - Call `unpack` on receiver side that will decrypt the message, verify signature if needed
+  and return a `Message` for further processing on the application level.
+
+### 1. Build an Encrypted DID Comm message for the given recipient
+
+This is most common DID Comm message to be used in most of the applications.
+
+A DIDComm encrypted message is an encrypted JWM (JSON Web Messages) and hides its content from all but authorized
```

Comment:
This sentence is a bit long and hard to read, I'd suggest bullet points such as:
```
A DIDComm encrypted message is an encrypted JWM (JSON Web Message) that:
- hides its content from all but the authorised recipients
- (optionally) discloses and proves the sender to _only_ those authorised recipients
- provides message integrity guarantees.

It is a safe format for storing DIDComm data at rest and is the usual format for transporting messages over a network.
```